### PR TITLE
build: Creating structure for CreateHotel

### PR DESCRIPTION
**Pull request by facamesi52**
## Summary
Creating the scaffolding of hotel entity.

- CRUD Endpoints

### CODE REVIEW

File: src/api/hotel/hotel.types.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    previous_price: number;
+    status: string;          
+    phone: string;                
+    hotel_type: string;
+    check_in: Date;           
+    check_out: Date;
+    rooms: any;     //preguntar
+    City_hotel: any; 
+    Service_labels_hotel: any;
+}*/
```
Comment (ht1204) [inline src/api/hotel/hotel.types.ts]: Please, remove dead or commented code which is not useful anymore.

File: src/api/hotel/index.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { Router } from "express";
+
+import { createHotelController } from "./hotel.controller";
+
+const router = Router();
+
+// /api/hotels -> POST
```
Comment (ht1204) [inline src/api/hotel/index.ts]: It's not necessary comments in this kind of implementations, i.e. this endpoint.

File: src/routes.ts [hunk 3]
```diff
@@ -5,10 +5,12 @@ import healthcheckRouter from './api/healthcheck';
...
 import userRouter from './api/user';
 // import authLocalRouter from './auth/local';
+import hotelRouter from './api/hotel';
 
 const routes = (app: Application) => {
   app.use('/api/healthcheck', healthcheckRouter)
   app.use('/api/user', userRouter)
+  app.use('api/hotel', hotelRouter)
 //   app.use('/api/products', productRouter)
 //   app.use('/api/reviews', reviewRouter)
```
Comment (ht1204) [inline src/routes.ts]: Please, remove commented code in this file.

File: src/api/hotel/hotel.controller.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+import { createHotel } from "./hotel.service";
+import { Hotel, CreateHotelData } from "./hotel.types";
+
+export async function createHotelController(req: Request, res: Response){
+    try{
+        const hotelData: CreateHotelData = req.body; 
+        const createdHotel: Hotel = await createHotel(hotelData);
+        res.status(201).json({ message: 'hotel has been created successfully',createdHotel});
+    } catch ({ message }: any) {
```
Comment (ht1204) [inline src/api/hotel/hotel.controller.ts]: I recommend create an importable resource for error management which contains a function like this:
```ts
function errorHandler(exception: unknown) {
  const message = (typeof exception === 'string') ? exception.toUpperCase()
    : (exception instanceof Error) ? exception.message : 'Something was wrong';

  return message;
}

export default errorHandler;
```

Then, in try catch implement something like this:
```ts
...
} catch(error: unknown) {
 const message = errorHandler(error);
 res.status(400).json({ message });
}
```

And do not forget semicolon once you use it in a project, thx.

Comment (ht1204) [reply src/api/hotel/hotel.controller.ts]: Apply this for a better error exception management instead of using any.

**Review by ht1204 — CHANGES_REQUESTED** [review]
Please, address the comments for enhancements. Thx in advance.

**Review by viamazoba — APPROVED** [review]

**Review by Nataliachm — APPROVED** [review]

**Pull request closed without merging**